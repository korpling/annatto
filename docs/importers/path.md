# path (importer)

Add file nodes for all files in the imported directory.

## Configuration

###  corpus_name



