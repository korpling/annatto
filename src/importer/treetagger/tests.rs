use std::path::Path;

use insta::assert_snapshot;

use crate::{importer::treetagger::TreeTaggerImporter, util::import_as_graphml_string};

const TT_DEFAULT_VIS_CONFIG: &str = r#"
[context]
default = 5
sizes = [0, 1, 2, 5, 10]

[view]
page_size = 10

[[visualizers]]
vis_type = "kwic"
display_name = "kwic"
visibility = "permanent"

[[visualizers]]
vis_type = "grid"
display_name = "grid"

[visualizers.mappings]
annos = "lb, highlighted"
"#;

#[test]
fn simple_token() {
    let actual = import_as_graphml_string(
        TreeTaggerImporter::default(),
        Path::new("tests/data/import/treetagger/token_only"),
        Some(TT_DEFAULT_VIS_CONFIG),
    )
    .unwrap();

    assert_snapshot!(actual);
}

#[test]
fn single_sentence() {
    let actual = import_as_graphml_string(
        TreeTaggerImporter::default(),
        Path::new("tests/data/import/treetagger/single_sentence"),
        Some(TT_DEFAULT_VIS_CONFIG),
    )
    .unwrap();

    assert_snapshot!(actual);
}
