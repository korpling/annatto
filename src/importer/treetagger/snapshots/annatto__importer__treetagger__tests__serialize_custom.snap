---
source: src/importer/treetagger/tests.rs
expression: serialization.unwrap()
---
column_names = ["a", "b", "c"]
file_encoding = "latin-1"
attribute_decoding = false
